File: Agente Yaiwes principal/_work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk-python/src/iii_sandbox/interpreter.py
from __future__ import annotations

from .client import HttpClient
from .types import CodeResult, KernelSpec


class CodeInterpreter:
    def __init__(self, client: HttpClient, sandbox_id: str):
        self._client = client
        self._sandbox_id = sandbox_id

    async def run(
        self, code: str, language: str = "python"
    ) -> CodeResult:
        data = await self._client.post(
            f"/sandbox/sandboxes/{self._sandbox_id}/interpret/execute",
            {"code": code, "language": language},
        )
        return CodeResult(**data)

    async def install(
        self, packages: list[str], manager: str = "pip"
    ) -> str:
        result = await self._client.post(
            f"/sandbox/sandboxes/{self._sandbox_id}/interpret/install",
            {"packages": packages, "manager": manager},
        )
        return result["output"]

    async def kernels(self) -> list[KernelSpec]:
        data = await self._client.get(
            f"/sandbox/sandboxes/{self._sandbox_id}/interpret/kernels"
        )
        return [KernelSpec(**k) for k in data]

File: Agente Yaiwes principal/_work/backend-watchdog-full-repair-02/src/iii-sandbox/packages/sdk/src/interpreter.ts
import type { HttpClient } from "./client.js"
import type { CodeResult, KernelSpec } from "./types.js"

export class CodeInterpreter {
  constructor(private client: HttpClient, private sandboxId: string) {}

  async run(code: string, language = "python"): Promise<CodeResult> {
    return this.client.post<CodeResult>(
      `/sandbox/sandboxes/${this.sandboxId}/interpret/execute`,
      { code, language },
    )
  }

  async install(packages: string[], manager: "pip" | "npm" | "go" = "pip"): Promise<string> {
    const result = await this.client.post<{ output: string }>(
      `/sandbox/sandboxes/${this.sandboxId}/interpret/install`,
      { packages, manager },
    )
    return result.output
  }

  async kernels(): Promise<KernelSpec[]> {
    return this.client.get<KernelSpec[]>(
      `/sandbox/sandboxes/${this.sandboxId}/interpret/kernels`,
    )
  }
}
